packages/@react-stately/menu/src/useSubMenuTriggerState.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {FocusStrategy} from '@react-types/shared';
+import {Key, useCallback, useMemo, useState} from 'react';
+import type {MenuTreeState} from './useMenuState';
+import type {OverlayTriggerState} from '@react-stately/overlays';
+import type {TreeState} from '@react-stately/tree';
+
+export interface SubMenuTriggerProps {
+  /** Key of the trigger item. */
+  triggerKey: Key
+}
+
+export interface SubMenuTriggerState extends OverlayTriggerState {
+  // TODO descriptions
+  // Need some form of isOpen and close for usePopover. Also need closeAll for onClose for menu items, would be kinda weird
+  // to pass in both menuTreeState and the state returned by useSubMenuTriggerState to useMenu.
+  isOpen: boolean,
+  focusStrategy: FocusStrategy | null,
+  open: (focusStrategy?: FocusStrategy | null) => void,
+  close: () => void,
+  closeAll: () => void,
+  // TODO: add setOpen and toggle for type parity with useOverlayTriggerState type that Tray and Popover expect. Perhaps call useOverlayTriggerState in
+  // this hook and just make it controlled? Then spread? Kinda weird since they are unused but needed for type parity.
+  toggle: (focusStrategy?: FocusStrategy | null) => void,
+  setOpen: () => void
+}
+
+// TODO: ideally onClose would be passed in from the Trigger level rather than the menu level so that we could call it
+// TODO descriptions and naming
+// TODO: I'm passing in the parent menu's TreeState now so that we can set expanded keys so a menu can know if it has any sub menus open.
+// The alternative to using expandedKeys would involve making each menu track which "level" it belongs to and have it the look up that index in
+// the MenuTreeState's expandedKeyStack and see if its collection has a matching key. This feels arguably more correct but would then involve
+// having useMenu accept MenuTreeState instead which might not be the worst thing tbh. Open to discussion
+export function UNSTABLE_useSubMenuTriggerState<T>(props: SubMenuTriggerProps, state: MenuTreeState & TreeState<T>): SubMenuTriggerState  {
```

Comment:
Open to opinions here, `useMenu`/Menu need to know if the menu has any expandedKeys for things like FocusScope containment and `aria-hidden`